### PR TITLE
allow oauth client to accept /oauth2 in base url

### PR DESCRIPTION
**Pull request by robrap**
The oauth client now accepts the base url of the LMS instance, with or
without including the path `/oauth2`.

The example value of either of the following two commonly found settings would
be acceptable as the `base_url` argument::

    LMS_BASE_URL = 'http://edx.devstack.lms:18000'
    BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL = 'http://edx.devstack.lms:18000/oauth2'

BOM-897

Note: Discussed adding this change with @nasthagiri before her vacation.

### CODE REVIEW

File: edx_rest_api_client/client.py [hunk 1]
```diff
@@ -91,26 +91,35 @@ class OAuthAPIClient(requests.Session):
...
     is subject to change.
     """
-    oauth_uri = '/oauth2/access_token'
-
     def __init__(self, base_url, client_id, client_secret, **kwargs):
         """
         Args:
-            base_url (str): base url of LMS instance
+            base_url (str): base url of LMS instance or
+                `BACKEND_SERVICE_EDX_OAUTH2_PROVIDER_URL` which already includes the path `/oauth2`.
```
Comment (dianakhuang) [inline edx_rest_api_client/client.py]: Maybe examples to clarify what you mean by this?

Comment (robrap) [reply edx_rest_api_client/client.py]: I’ll add that next year. 😉

Comment (robrap) [reply edx_rest_api_client/client.py]: Done.

Comment (robrap) [reply edx_rest_api_client/client.py]: Done.

**Pull request merged**